stratuslab/python/tar/slipstream_stratuslab/StratusLabClientCloud.py

```diff
@@ -365,12 +365,17 @@ def _stop_deployment(self):
 
     @override
     def _stop_vms_by_ids(self, ids):
+        """ids : list of IaaS instance IDs to stop.
+        Returns list of the instance IDs for which the termination call succeeded.
+        """
         runner = self._get_stratuslab_runner(None, self.slConfigHolder.copy())
         for _id in map(int, ids):
             try:
                 runner.killInstances([_id])
             except Exception as ex:
                 self._print_detail("Failed to terminate VM %s: %s" % (_id, str(ex)))
+                ids.remove(_id)
```

Comment:
I would create a deep copy of `ids` to prevent some unexpected behaviour if the list of ids (is|will be) used by the caller.
@konstan : What do you think ?